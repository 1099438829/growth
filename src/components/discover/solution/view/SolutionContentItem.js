import React, { Component, PropTypes } from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import AppStyle from '../../../../theme/styles';
import Line from '../../../../components/Line';

class SolutionContentItem extends Component {
  static componentName = 'SolutionContentItem';

  static propTypes = {
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    stack: PropTypes.arrayOf(
      PropTypes.string,
    ).isRequired,
    click: PropTypes.func,
  };

  static defaultProps = {
    name: '',
    description: '',
    stack: [],
    click: () => {},
  };

  render() {
    const stack = this.props.stack
      .map((val, index) => (
        <Text key={val.concat(index)}>{val}</Text>));
    return (
      <TouchableHighlight
        style={AppStyle.solutionContentItemStyle}
        onPress={this.props.click}
        underlayColor="rgba(250, 250, 250, .9)"
      >
        <View>
          <View style={AppStyle.solutionContentItemTitleStyle}>
            <Text>{this.props.name}</Text>
          </View>
          <Text style={AppStyle.solutionContentItemDescriptionStyle}>{this.props.description}</Text>
          <Line />
          <View style={AppStyle.solutionContentItemStackStyle}>
            {stack}
          </View>
        </View>
      </TouchableHighlight>
    );
  }
}
export default SolutionContentItem;
