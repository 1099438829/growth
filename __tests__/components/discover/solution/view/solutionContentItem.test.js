import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import SolutionContentItem from '../../../../../src/components/discover/solution/view/SolutionContentItem';
import SOLUTIONS from '../../../../../src/constants/SOLUTIONS';

it('renders correctly', () => {
  const data = SOLUTIONS[0];
  const solutionContentItem = renderer.create(
    <SolutionContentItem
      name={data.name}
      description={data.description}
      stack={data.stacks}
    />,
  );
  const solutionContentItemJson = solutionContentItem.toJSON();
  expect(solutionContentItemJson).toMatchSnapshot();
});
