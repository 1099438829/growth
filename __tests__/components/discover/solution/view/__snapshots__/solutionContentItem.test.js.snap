// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "white",
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 8,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Nginx/Apache日志分析
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "padding": 10,
        }
      }
    >
      用Hadoop分析、解析数据，将数据存储到ElasticSearch中，通过API与amMap再显示到地图上。
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "rgba(240, 240, 240, .9)",
          "height": 1,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginBottom": 5,
          "marginTop": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Hadoop
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        amMap
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        ElasticSearch
      </Text>
    </View>
  </View>
</View>
`;
